python/cuopt/cuopt/linear_programming/problem.py

```diff
@@ -0,0 +1,384 @@
+        match other:
+            case int() | float():
+                return Constraint(self, CONSTRAINT_LE, float(other))
+            case Variable() | LinearExpression():
+                # expr1 <= expr2   -> expr1 - expr2 <= 0
+                expr = self - other
+                return Constraint(expr, CONSTRAINT_LE, 0.0)
+    def __ge__(self, other):
+        match other:
+            case int() | float():
+                return Constraint(self, CONSTRAINT_GE, float(other))
+            case Variable() | LinearExpression():
+                # expr1 >= expr2   ->  expr1 - expr2 >= 0
+                expr = self - other
+                return Constraint(expr, CONSTRAINT_GE, 0.0)
+    def __eq__(self, other):
+        match other:
+            case int() | float():
+                return Constraint(self, CONSTRAINT_EQ, float(other))
+            case Variable() | LinearExpression():
+                # expr1 == expr2   -> expr1 - expr2 == 0
+                expr = self - other
+                return Constraint(expr, CONSTRAINT_EQ, 0.0)
+
+# The sense of a constraint is either less than or equal, greater than or equal, or equal
+CONSTRAINT_LE = 'L'
+CONSTRAINT_GE = 'G'
+CONSTRAINT_EQ = 'E'
+
+# A constraint contains a linear expression, the sense of the constraint, and the right-hand side of the constraint
+class Constraint:
+    def __init__(self, expr, sense, rhs, name=''):
+        self.vindex_coeff_dict = {}
+        nz = len(expr.getVars())
+        self.vars = expr.vars
+        for i in range(nz):
+            v_idx = expr.vars[i].index
+            v_coeff = expr.coefficients[i]
+            self.vindex_coeff_dict[v_idx] = self.vindex_coeff_dict[v_idx] + v_coeff if v_idx in self.vindex_coeff_dict else v_coeff
+        self.Sense = sense
+        self.RHS = rhs - expr.getConstant()
+        self.CName = name
+    def getName(self):
+        return CName
+    def getSense(self):
+        return self.Sense
+    def getRHS(self):
+        return self.RHS
+    def getCoefficient(self, var):
+        v_idx = var.index
+        return vindex_coeff_dict[v_idx]
+    @property
+    def Slack(self):
+        lhs = 0.0
+        for var in self.vars:
+            lhs += var.Value * self.vindex_coeff_dict[var.index]
+        return self.RHS - lhs
+
+# The sense of a problem is either minimize or maximize
+MINIMIZE = 0
+MAXIMIZE = 1
+
+# A Problem defines a Linear Program or Mixed Integer Program
+# Variable can be be created by calling addVariable()
+# Constraints can be added by calling addConstraint()
+# The objective can be set by calling setObjective()
+# The problem data is formed when calling optimize()
+class Problem:
+    def __init__(self, model_name=''):
+        self.Name = model_name
+        self.vars = []
+        self.constrs = []
+        self.ObjSense = MINIMIZE
+        self.Obj = None
+        self.ObjCon = 0.0
+        self.Status = -1
+        self.IsMIP = False
+        self.Settings = solver_settings.SolverSettings()
```

Comment:
A function to set solver settings

Reply:
Moved settings as a parameter to the problem.